flink-runtime-web/web-dashboard/src/app/pages/job-manager/thread-dump/job-manager-thread-dump.component.ts

```diff
@@ -31,16 +35,26 @@ import { EditorOptions } from 'ng-zorro-antd/code-editor/typings';
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class JobManagerThreadDumpComponent implements OnInit, OnDestroy {
-  public readonly editorOptions: EditorOptions = flinkEditorOptions;
-
+  public editorOptions: EditorOptions;
   public dump = '';
   public loading = true;
+  public downloadUrl = '';
+  public downloadName = '';
 
   private readonly destroy$ = new Subject<void>();
 
-  constructor(private readonly jobManagerService: JobManagerService, private readonly cdr: ChangeDetectorRef) {}
+  constructor(
+    private readonly jobManagerService: JobManagerService,
+    private readonly configService: ConfigService,
+    private readonly cdr: ChangeDetectorRef,
+    @Inject(JOB_MANAGER_MODULE_CONFIG) readonly moduleConfig: JobManagerModuleConfig
+  ) {
+    this.editorOptions = moduleConfig.editorOptions || JOB_MANAGER_MODULE_DEFAULT_CONFIG.editorOptions;
+  }
 
   public ngOnInit(): void {
+    this.downloadUrl = `${this.configService.BASE_URL}/jobmanager/thread-dump`;
+    this.downloadName = `jobmanager_thread_dump`;
```

Comment:
These can be assigned at their declaration directly (and could be made `readonly` then).